dataset_loaders/parallel_loader.py

```diff
@@ -637,6 +638,17 @@ def fetch_from_dataset(self, batch_to_load):
                 if self.set_has_GT and self.return_one_hot:
                     seq_y = seq_y.transpose([0, 3, 1, 2])
                 raw_data = raw_data.transpose([0, 3, 1, 2])
+                if seq_x.shape[3] == 1:
+                    seq_x = seq_x.squeeze(3)
+                    seq_y = seq_y.squeeze(3) if self.return_one_hot else \
+                        seq_y.squeeze(2)
+                    raw_data = raw_data.squeeze(3)
+            else:
+                if seq_x.shape[2] == 1:
+                    seq_x = seq_x.squeeze(2)
+                    seq_y = seq_y.squeeze(2) if self.return_one_hot else \
+                        seq_y.squeeze(1)
+                    raw_data = raw_data.squeeze(2)
```

Comment:
If you modify the parallel_loaders to accept 3D data out of `load_sequence` you shouldn't need all these modifications.